src/parameterizations/lateral/MOM_spherical_harmonics.F90

```diff
@@ -240,8 +253,13 @@ subroutine spherical_harmonics_init(G, param_file, CS)
   allocate(CS%a_recur(CS%ndegree+1, CS%ndegree+1)); CS%a_recur(:,:) = 0.0
   allocate(CS%b_recur(CS%ndegree+1, CS%ndegree+1)); CS%b_recur(:,:) = 0.0
   do m=0,CS%ndegree ; do n=m+1,CS%ndegree
+    ! This expression will give NaNs with 32-bit integers for n > 23170, but this is trapped elsewhere.
     CS%a_recur(n+1,m+1) = sqrt(real((2*n-1) * (2*n+1)) / real((n-m) * (n+m)))
-    CS%b_recur(n+1,m+1) = sqrt(real((2*n+1) * (n+m-1) * (n-m-1)) / real((n-m) * (n+m) * (2*n-3)))
+    if (n <= 1024) then
+      CS%b_recur(n+1,m+1) = sqrt(real((2*n+1) * (n+m-1) * (n-m-1)) / real((n-m) * (n+m) * (2*n-3)))
```

Comment:
Yes, the small n branch is there to retain old answers.  It is possible that just using the second branch would give the same answers, but I have not tested this.

Reply:
I just made a quick comparison with my tidal run and it did not change answer if the small n branch is removed, so either way is fine with me.

Reply:
Thanks for doing that check that the revised version gives identical answers in all cases.  I have updated the PR to avoid the separate large-n and small-n branches and simplify the code.